src/editors/documentController.ts

```diff
@@ -0,0 +1,262 @@
+    const activeEditor = vscode.window.activeTextEditor;
+
+    log.info('save document to MongoDB', activeEditor);
+
+    if (!activeEditor) {
+      return false;
+    }
+
+    const uriParams = new URLSearchParams(activeEditor.document.uri.query);
+    const documentLocation =
+      uriParams.get(DOCUMENT_LOCATION_URI_IDENTIFIER) || '';
+    const namespace = uriParams.get(NAMESPACE_URI_IDENTIFIER);
+    const connectionId = uriParams.get(CONNECTION_ID_URI_IDENTIFIER);
+    const documentIdReference = uriParams.get(DOCUMENT_ID_URI_IDENTIFIER) || '';
+    const documentId = this._documentIdStore.get(documentIdReference);
+
+    // If not MongoDB document save to disk instead of MongoDB.
+    if (
+      documentLocation !== 'mongodb' ||
+      !namespace ||
+      !connectionId ||
+      !documentId
+    ) {
+      vscode.commands.executeCommand('workbench.action.files.save');
+
+      return false;
+    }
+
+    const activeConnectionId = this._connectionController.getActiveConnectionId();
+    const connectionName = this._connectionController.getSavedConnectionName(
+      connectionId
+    );
+
+    if (activeConnectionId !== connectionId) {
+      return this._saveDocumentFailed(
+        `Unable to save document: no longer connected to '${connectionName}'`
+      );
+    }
+
+    const dataservice = this._connectionController.getActiveDataService();
+
+    if (dataservice === null) {
+      return this._saveDocumentFailed(
+        `Unable to save document: no longer connected to '${connectionName}'`
+      );
+    }
+
+    return await this.replaceDocument(
+      documentId,
+      namespace,
+      dataservice,
+      activeEditor
+    );
+  }
+
+  async replaceDocument(
+    documentId: EJSON.SerializableTypes,
+    namespace: string,
+    dataservice: DataServiceType,
+    editor: vscode.TextEditor
+  ): Promise<boolean> {
+    const activeEditor = vscode.window.activeTextEditor;
+    let newDocument: EJSON.SerializableTypes = {};
+
+    try {
+      newDocument = EJSON.parse(editor.document.getText());
+    } catch (error) {
+      return this._saveDocumentFailed(error.message);
+    }
+
+    this._statusView.showMessage('Saving document...');
+
+    dataservice.findOneAndReplace(
+      namespace,
+      {
+        _id: documentId
+      },
+      newDocument,
+      {
+        returnOriginal: false
+      },
+      (error, data) => {
+        this._statusView.hideMessage();
+
+        if (error) {
+          return this._saveDocumentFailed(
+            `Unable to save document: ${error.message}`
+          );
+        }
+
+        // Send metrics to Segment.
+        this._telemetryController.trackDocumentUpdated('treeview', true);
+
+        // Save document changes to active editor.
+        activeEditor?.document.save();
+
+        // Update parent list of documents to reflect the changes that were made.
+        vscode.commands.executeCommand(
+          EXTENSION_COMMANDS.MDB_REFRESH_PLAYGROUND_RESULT_CONTENT,
```

Comment:
A bit of a functionality question - do we want to update the documents that were returned from the result of a playground here? I think it might be a little counter intuitive for the playground results to be updated when a document is updated. Would just the updated document be changed? What happens if other documents provided in the `find` of the playground results were updated in the background?

Reply:
According to the acceptance criteria, we should update the parent list: `When the document is saved (Cmd/Ctrl + S or with the menu), the document in the collection is updated to reflect the changes that were made`

https://jira.mongodb.org/browse/VSCODE-178

But let's talk about it at the dosing sync.